vcf_annotation_tools/vep_annotation_reporter.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import argparse
+import re
+import vcfpy
+import tempfile
+import csv
+import binascii
+
+def define_parser():
+    parser = argparse.ArgumentParser('vep-annotation-reporter')
+    parser.add_argument(
+        "input_vcf",
+        help="The VCF file with VEP annotations to report."
+    )
+    parser.add_argument(
+        "vep_fields",
+        help="The VEP fields to report.",
+        nargs='+',
+    )
+    parser.add_argument(
+        "-t", "--input_tsv",
+        help="A TSV report file to add VEP annotations to. Required columns are CHROM, POS, REF, ALT. These are used to match each TSV entry to a VCF entry. Must be tab-delimited."
+    )
+    parser.add_argument(
+        "-o", "--output-tsv",
+        help="Path to write the output report TSV file. If not provided, the output TSV will be written "
+            +"next to the input VCF with a .tsv file ending."
+    )
+    return parser
+
+def create_vcf_reader(args):
+    vcf_reader = vcfpy.Reader.from_path(args.input_vcf)
+    if 'CSQ' not in vcf_reader.header.info_ids():
+        vcf_reader.close()
+        raise Exception("ERROR: VCF {} is not VEP-annotated. Please annotate the VCF with VEP before running this tool.".format(args.input_vcf))
+    return vcf_reader
+
+def create_tsv_reader(input_filehandle):
+    tsv_reader = csv.DictReader(input_filehandle, delimiter = "\t")
+    for field in ['CHROM', 'POS', 'REF', 'ALT']:
+        if field not in tsv_reader.fieldnames:
+            raise Exception("ERROR: Input TSV {} doesn't contain required column '{}'.".format(input_filehandle.name, field))
+    return tsv_reader
+
+def parse_csq_header(vcf_reader):
+    format_pattern = re.compile('Format: (.*)')
+    return format_pattern.search(vcf_reader.header.get_info_field_info('CSQ').description).group(1).split('|')
+
+def parse_csq_entries(csq_entries, csq_fields):
+    transcripts = {}
+    for entry in csq_entries:
+        values = entry.split('|')
+        transcript = {}
+        for key, value in zip(csq_fields, values):
+            transcript[key] = value
+        if transcript['Allele'] not in transcripts.keys():
+            transcripts[transcript['Allele']] = []
+        transcripts[transcript['Allele']].append(transcript)
+    return transcripts
+
+def is_sv(entry):
+    return 'SVTYPE' in entry.INFO
+
+def resolve_alleles(entry, csq_alleles):
+    alleles = {}
+    if is_sv(entry):
+        for alt in entry.ALT:
+            alt = alt.serialize()
+            if len(alt) > len(entry.REF) and 'insertion' in csq_alleles:
+                alleles[alt] = 'insertion'
+            elif len(alt) < len(entry.REF) and 'deletion' in csq_alleles:
+                alleles[alt] = 'deletion'
+            elif len(csq_alleles) == 1:
+                alleles[alt] = list(csq_alleles)[0]
```

Comment:
Is it possible to end up in an `is_sv` case where none of these conditions are true?

Reply:
Possibly but from what I remember back when I first added this case, VEP seems to annotate sv variants as either `insertion` or `deletion`. I added this the case because it would catch most of the cases if the above assumption doesn't hold true overall. That case works as long as the site has a single alt allele. It's not clear how VEP would encode the allele otherwise since I haven't encountered that yet. It might make sense to add a warning message for the remaining cases.